File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        countDict = {}
        
        for n in nums:
            if n in countDict:
                countDict[n] += 1
                return True
            else:
                countDict[n] = 1
        
        return False
        

File: 0217-contains-duplicate/0217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Map<Integer, Integer> numMap = new HashMap<>();

        for(int num: nums) {
            if (numMap.get(num) != null) return true;
            numMap.put(num, 1);
        }
        return false;
    }
}